frontend/src/pages/distributedWorkloads/global/workloadStatus/DWStatusOverviewDonutChart.tsx

```diff
@@ -1,23 +1,31 @@
 import * as React from 'react';
 import { Card, CardTitle, CardBody, Bullseye, Spinner } from '@patternfly/react-core';
+import { ChartDonut, ChartThemeColor } from '@patternfly/react-charts';
 import EmptyStateErrorMessage from '~/components/EmptyStateErrorMessage';
 import { DistributedWorkloadsContext } from '~/concepts/distributedWorkloads/DistributedWorkloadsContext';
+import {
+  WorkloadStatusColorAndIcon,
+  WorkloadStatusType,
+  getStatusCounts,
+} from '~/concepts/distributedWorkloads/utils';
 
 export const DWStatusOverviewDonutChart: React.FC = () => {
-  const { workloadCurrentMetrics } = React.useContext(DistributedWorkloadsContext);
+  const { workloads } = React.useContext(DistributedWorkloadsContext);
 
-  if (workloadCurrentMetrics.error) {
+  const statusCounts = React.useMemo(() => getStatusCounts(workloads.data), [workloads.data]);
```

Comment:
FYI, just adding the thing we discussed on the peer review. We are memoizing here a k8s object, bear in mind that state changes on complex objects like this wont bring anything relevant into the table, since pretty much all the refecth will get a different reference, maybe we'll wanna revisit this for a performance standpoint in the future

Reply:
(Just repeating my thoughts from the meeting as well so I remember in the future 😆)
Yeah, that's a fair point. It's probably an overoptimization to begin with because `getStatusCounts` isn't that heavy a calculation, I just wanted to prevent it from getting re-run even when the data hasn't been refetched (if the tree re-renders for a different reason). The way to memoize it correctly though would be to only recalculate it if the `status.conditions` array on any workload has changed, and checking that via `useDeepCompareMemoize` would be as much (or more) computation than just re-running `getStatusCounts` I think